localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1865,12 +1857,7 @@ def create_event_source_mapping(
         context: RequestContext,
         request: CreateEventSourceMappingRequest,
     ) -> EventSourceMappingConfiguration:
-        if config.LAMBDA_EVENT_SOURCE_MAPPING == "v2":
-            event_source_configuration = self.create_event_source_mapping_v2(context, request)
-        else:
-            event_source_configuration = self.create_event_source_mapping_v1(context, request)
-
-        return event_source_configuration
+        return self.create_event_source_mapping_v2(context, request)
```

Comment:
Should this function not just contain the contents of `create_event_source_mapping_v2` -- with `create_event_source_mapping_v2` being removed?

Will probably be the same for some other functions like `update_event_source_mapping_v2` so could be worthwhile to do in a follow-up.

Reply:
+1 for cleaning up in a follow up and keeping these changes here minimal